sky/packages/sky/lib/painting/box_painter.dart

```diff
@@ -516,9 +516,15 @@ class BoxPainter {
     if (_decoration.border == null)
       return;
 
-    if (_hasUniformBorder && _decoration.borderRadius != null) {
-      _paintBorderWithRadius(canvas, rect);
-      return;
+    if (_hasUniformBorder) {
+      if (_decoration.borderRadius != null) {
+        _paintBorderWithRadius(canvas, rect);
+        return;
+      }
+      if (_decoration.shape == Shape.circle) {
+        _paintBorderWithCircle(canvas, rect);
+        return;
+      }
```

Comment:
change the TODO(ianh) comment on the assert a bit lower down to say "Support _non-uniform_ borders on circles", since we now do support them if they're uniform.